src/interpreter/mod.rs

```diff
@@ -757,7 +757,8 @@ where
     F: FnOnce(&bitcoin::PublicKey, BitcoinSig) -> bool,
 {
     if let Some((sighash_byte, sig)) = sigser.split_last() {
-        let sighashtype = bitcoin::SigHashType::from_u32(*sighash_byte as u32);
+        let sighashtype = bitcoin::SigHashType::from_u32_standard(*sighash_byte as u32)
```

Comment:
I wonder if the interpreter should use the `from_u32_consensus`. It is used to interpret the transactions that are already present in the blockchain. The main use is for block explorers which should accept non-standard transactions

Reply:
Hmm, but i thought we shouldn't be parsing invalid-by-standardness Script? Or is it called by `parse_insane_*`?

Reply:
The goal of `parse_insane` was to parse non-safe miniscripts or miniscripts containing duplicate keys or miniscripts possibly exceeding resource limits while satisfaction. 

`parse_insane` does not currently support parsing non-standard scripts, so what you are doing in this PR seems correct. Having a non-standard miniscript is out of scope for now. This is can be addressed later(while we address #165)